src/string/Base64.ts

```diff
@@ -0,0 +1,136 @@
+/**
+ * Representing a Base64 encoded string. Strings are URL-safe by default, but can be made
+ * non-safe by providing an option to the constructor, e.g...
+ *
+ * ```ts
+ * const decoder = Base64.Decoder({ urlSafe: false })
+ * ```
+ *
+ * This module is heavily inspired by the `validator.js` module
+ * [`isBase64`](https://github.com/validatorjs/validator.js/blob/master/src/lib/isBase64.js).
+ *
+ * @since 0.0.2
+ */
+import { Kind, Kind2, URIS, URIS2, HKT } from 'fp-ts/HKT'
+import * as D from 'io-ts/Decoder'
+import * as Eq_ from 'fp-ts/Eq'
+import * as G from 'io-ts/Guard'
+import * as Str from 'fp-ts/string'
+import * as TD from 'io-ts/TaskDecoder'
+import * as t from 'io-ts/Type'
+import { pipe } from 'fp-ts/function'
+
+/**
+ * @since 0.0.2
+ * @category Internal
+ */
+interface Base64Brand {
+  readonly Base64: unique symbol
+}
+
+/**
+ * Representing a Base64 encoded string.
+ *
+ * Heavily inspired by the `validator.js` module
+ * [`isBase64`](https://github.com/validatorjs/validator.js/blob/master/src/lib/isBase64.js).
+ *
+ * @since 0.0.2
+ * @category Model
+ */
+export type Base64 = string & Base64Brand
+
+/**
+ * @since 0.0.1
+ * @category Model
+ */
+export type Base64SchemableOptions = {
+  urlSafe: boolean
+}
+
+/**
+ * @since 0.0.1
+ * @category Model
+ */
+export type SchemableParams<S> = (options?: Base64SchemableOptions) => HKT<S, Base64>
+
+/**
+ * @since 0.0.1
+ * @category Model
+ */
+export type SchemableParams1<S extends URIS> = (
+  options?: Base64SchemableOptions
```

Comment:
💬  I'm open to optional params, but I ended up choosing against it for UUID because otherwise it doesn't make much sense why there's a function call for the combinator:

```typescript
const User = SC.make(S => 
  S.struct({
    token: S.Base64()
  })
)
```

vs an explicit params, which make it clearer why it's a function call:

```typescript
const User = SC.make(S => 
  S.struct({
    token: S.Base64({ urlSafe: true })
  })
)
```

And like you mentioned, an alternative for these simple configs would be to split them up into different modules

Reply:
I ended up adding an additional module. I'll open a new issue to cover this. And it's probably worth discussing this some more since I expect we'll need to find out a middle ground between representing all possible variants, and having an API that is reasonable compact and usable for most use cases.

Reply:
Created #44